File: LeetcodePython/101.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSym(self, p, q):
        if not p and not q:
            return True
        if p and q and p.val == q.val:
            return self.isSym(p.left, q.right) and self.isSym(p.right, q.left)
        return False


    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root:
            return self.isSym(root.left, root.right)
        return True

File: LeetcodeCPP/101.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if (!root) return true;
        else return isSym(root->left, root->right);
    }
    
    bool isSym(TreeNode *lr, TreeNode *rr) {
        if(!lr && !rr) return true;
        if (lr && rr && lr->val == rr->val){
            return isSym(lr->left, rr->right) && isSym(lr->right, rr->left);
        }
        return false;
    }
};